interfaces/builtin/browser_support.go

```diff
@@ -279,7 +279,7 @@ dbus (send)
     bus=system
     path=/org/freedesktop/RealtimeKit1
     interface=org.freedesktop.RealtimeKit1
-    member=MakeThread{HighPriority,Realtime}
+    member=MakeThread{HighPriority,Realtime,RealtimeWithPID}
```

Comment:
Looking at https://github.com/heftig/rtkit/blob/c295fa849f52b487be6433e69e08b46251950399/rtkit-daemon.c `MakeThreadRealtimeWithPID` is just an extension of MakeThreadHighPriority which allows to pass the PID as a parameter (instead of the implicit PID of the caller). Both code paths verify that the process indeed belongs to the user on whose behalf the call was made, so I think it's ok to allow this call. (actually both calls are verified in the same code path).

Reply:
Ideally we would restrict a snap to changing only it's own processes/threads priorities - but in this case we would allow a snap to change either another snaps processes or a non-snap process. In which case I wonder if this is better suited to to `process-control` interface which already has similar semantics. Or we add this but then make `browser-support` not be auto-connected as it currently is to reduce the impact of this as a possible security issue?

Reply:
Isn't there a way to filter out processes/threads that don't belong to the caller?

Reply:
@oSoMoN unfortunately there is not, this is a long standing issue with AppArmor that there is not such a mechanism to only allow certain pid's, the only mechanism is "owner" which is the same uid (i.e. root controls root processes, user controls that user processes etc) and not by the apparmor label of that process/PID

Reply:
Personally, while I would prefer that this sort of access be in process-control, I think this is okay for the browser-support interface with the `allow-sandbox: true` attribute specifically because:

1) browsers using browser-support with `allow-sandbox: true` already have more powerful powers granted to them in the form of the `sys_admin`, `chown`, `fsetid`, `setgid`, `setuid`, and `sys_chroot` capabilities - these things in combination with the fact that browser snaps are permitted to ship setuid binaries for enforcing browser sandboxes, so this isn't a meaningful additional privilege, it just perhaps makes certain kinds of attacks easier.
1) having firefox instead plug `process-control` grants much more permission to it than just adding this here
1) usage of the `allow-sandbox: true` variant of process-control requires store approval

Reply:
Ah that is a good point - I had forgotten about `allow-sandbox: true` - the use of this in the store is guarded by a store declaration too and you are right that in general `process-control` is probably too much for firefox et al. (I still think this should perhaps *also* live in `process-control` too FWIW).